subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccPlatformToolProvider.java

```diff
@@ -79,8 +88,13 @@
     @Override
     protected Compiler<CCompileSpec> createCCompiler() {
         GccCommandLineToolConfigurationInternal cCompilerTool = toolRegistry.getTool(ToolType.C_COMPILER);
+        GccVersionResult gccMetadata = gccMetadata(cCompilerTool);
         CCompiler cCompiler = new CCompiler(buildOperationExecutor, compilerOutputFileNamingSchemeFactory, commandLineTool(cCompilerTool), context(cCompilerTool), getObjectFileExtension(), useCommandFile, workerLeaseService);
-        return new OutputCleaningCompiler<CCompileSpec>(cCompiler, compilerOutputFileNamingSchemeFactory, getObjectFileExtension());
+        return new VersionedOutputCleaningCompiler<CCompileSpec>(cCompiler, compilerOutputFileNamingSchemeFactory, getObjectFileExtension(), compilerType(gccMetadata), gccMetadata.getVersion());
+    }
+
+    private String compilerType(GccVersionResult gccMetadata) {
+        return gccMetadata.isClang() ? ClangToolChain.DEFAULT_NAME : GccToolChain.DEFAULT_NAME;
```

Comment:
Is it better to track a simple string, or should we track the type of the toolchain (i.e. the FQCN of the toolchain)?

Reply:
I am not so sure, yet. We somehow want to identify an executable of a toolchain. To differ between the Clang/Gcc and Visual Studio toolchains we can use the class of the toolchain or the name. What would be the advantage of using the FQCN of the toolchain?
There are still more questions about what to track as `type`: Do we want to make a difference between `clang` and `clang++` or `gcc` and `g++`? Do we want to differ between executables for a different architectures? I guess I'll need to ask the native team.

Reply:
Sterling and I discussed this a little, and I think we're probably ok with what we have.  If it's the same toolchain/version and the same platform, library inputs, etc, etc, etc, then I think that's enough.